docs/proposals/proposal-workflow-authentication.md

```diff
@@ -0,0 +1,69 @@
+# Proposal for Workflow Authentication
+
+## Proposal summary
+
+**For workflow-to-workflow-service authentication, use:**
+
+* **for jobs: one-time per-job tokens**, i.e. a single one-time token for
+  every job;
+* **for the user/interactive CLI: timed tokens**.
+
+Per-job tokens would require one of the following (or another alternative):
+
+  * another file per job, which is not ideal as it adds to the file count;
+  * a shared combined file containing tokens for all jobs per workflow, also
+    not ideal as it means that jobs would needlessly know about other jobs'
+    tokens;
+  * extending an existing job file such as the 'job.status' so that these files
+    also carry the token information for each job, & preventing that
+    information from being displayed in Cylc Review.
+
+Decisions on implementation details, such as modules & standards to use,
+await the go-ahead on an approach, the above one or otherwise.
+
+
+## Further information
+
+### Context
+
+Cylc 8 discussions have converged (as described briefly
+[here](cylc-8-architecture#command-line-interface) &
+[here](cylc-8-tasks#general-authentication-issues-sujata-hilary--damian-and-martin-)
+on usage of tokens for authentication between any workflow & the workflow
```

Comment:
Again, I'm a bit confused by the terminology here "any *workflow* & the workflow service"??